Python/Product/Analysis/PythonAnalyzer.cs

```diff
@@ -344,133 +344,14 @@ public partial class PythonAnalyzer : IGroupableAnalysisProject, IDisposable {
-                    if (string.IsNullOrEmpty(importingFrom)) {
-                        importingFrom = "__init__";
-                    } else {
-                        importingFrom += ".__init__";
-                    }
-                }
-            }
-
-            if (string.IsNullOrEmpty(relativeModuleName)) {
-                yield break;
-            }
-
-            // Handle relative module names
-            if (relativeModuleName.FirstOrDefault() == '.') {
-                if (string.IsNullOrEmpty(importingFrom)) {
-                    // No source to import relative to.
-                    yield break;
-                }
-
-                var prefix = importingFrom.Split('.');
-
-                if (relativeModuleName.All(c => c == '.')) {
-                    // The whole name is dots, so there's nothing to concatenate.
-                    yield return string.Join(".", prefix.Take(prefix.Length - relativeModuleName.Length));
-                } else {
-                    // Assume trailing dots are not part of the import
-                    var suffix = relativeModuleName.TrimEnd('.').Split('.');
-                    var dotCount = suffix.TakeWhile(bit => string.IsNullOrEmpty(bit)).Count();
-                    if (dotCount < prefix.Length) {
-                        // If we have as many dots as prefix parts, the entire
-                        // name will disappear. Despite what PEP 328 says, in
-                        // reality this means the import will fail.
-                        yield return string.Join(".", prefix.Take(prefix.Length - dotCount).Concat(suffix.Skip(dotCount)));
-                    }
-                }
-                yield break;
-            }
-
-            // The two possible names that can be imported here are:
-            // * relativeModuleName
-            // * importingFrom.relativeModuleName
-            // and the order they are returned depends on whether
-            // absolute_import is enabled or not.
-
-            // Assume trailing dots are not part of the import
-            relativeModuleName = relativeModuleName.TrimEnd('.');
-
-            // With absolute_import, we treat the name as complete first.
-            if (absoluteImports) {
-                yield return relativeModuleName;
-            }
-
-            if (!string.IsNullOrEmpty(importingFrom)) {
-                var prefix = importingFrom.Split('.');
-
-                if (prefix.Length > 1) {
-                    var adjacentModuleName = string.Join(".", prefix.Take(prefix.Length - 1)) + "." + relativeModuleName;
-                    yield return adjacentModuleName;
-                }
-            }
-
-            // Without absolute_import, we treat the name as complete last.
-            if (!absoluteImports) {
-                yield return relativeModuleName;
+            var unresolved = importFrom.GetModuleInfo().GetAllUnresolvedModules();
+            if(unresolved.Count == 0) {
```

Comment:
Minor spacing issues all over the place - suggest Ctrl+K, Ctrl+D everywhere before merging (and check that it hasn't broken anything when you do that).